[Keychron C1] Bulk update

vjdato21 opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
Addresses known issues and aligns readme page to Keychron's official readme page.
<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* Applied a workaround for the USB descriptor
* Fixed an issue where the keyboard gets detected as a 6KRO keyboard instead of NKRO keyboard.
* Updated keymapping to reflect the stock keymap of Keychron C1.
* Fixed Siri hotkey.
* Updated readme to reflect or mimic Keychron's official readme.
* Fixed status LEDs.
* `KEYBOARD_SHARED_EP` is now used to optimize for the lack of endpoints in SN32F248B.
* Keyboard may not send keystrokes or be detected if it was plugged to the host PC beforehand. Use the restart keyboard button combination `Fn + B` to work around this.

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

vjdato21 committed:
[Keychron C1] Bulk update

- Fixed USB descriptor
- Improved keyboard responsiveness.
- Fixed an issue where the keyboard gets detected as a 6KRO keyboard instead of NKRO keyboard.
- Updated keymapping to reflect the stock keymap of Keychron C1.
- Fixed Siri hotkey.
- Updated readme to reflect or mimic Keychron's official readme.

simgunz commented on keyboards/keychron/c1/keymaps/default/keymap.c:
```diff
@@ -55,59 +63,58 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+                {   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS },
+                {   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_NO,     KC_TRNS,   KC_NO,     KC_NO,     KC_NO   },
+                {   KC_TRNS,   KC_NO,     KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_NO,     KC_TRNS,   KC_NO,     RGB_SPI,   KC_NO   },
+                {   KC_TRNS,   KC_LALT,   KC_LGUI,   KC_NO,     KC_NO,     KC_NO,     KC_TRNS,   KC_NO,     KC_NO,     KC_NO,     KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   RGB_SAI,   RGB_SPD,   RGB_SAD }
```
KC_LALT and KC_LGUI are wrongly inverted respect the base windows layer (layer 2), shouldn't be they set to kc_trans?

vjdato21 replied:
Oh yeah, my bad. Will change it.

vjdato21 committed:
Update keymap.c

vjdato21 committed:
Update keymap.c

vjdato21 committed:
Update rules.mk

vjdato21 committed:
Fix descriptor better

- IsaacDynamo suggested to use '\x43\x31' instead as it means 'C1' in ASCII hex.

vjdato21 committed:
Attempt to fix status LEDs

- Added IsaacDynamo's work/suggestion.

vjdato21 committed:
Update config.h

vjdato21 committed:
Added back reset/restart keyboard custom keycode

vjdato21 committed:
Update keymap.c

vjdato21 committed:
Update readme.md

vjdato21 committed:
Align config.h to the changes in keymap.c

- Changed `LED_SCROLL_LOCK_PIN` pin to `LED_MAC_PIN`.
- Changed `LED_NUM_LOCK_PIN` to `LED_WIN_PIN`.

vjdato21 committed:
Update config.h

vjdato21 committed:
Optimize for the amount of EP in the MCU

- Fixes an issue where they keyboard may become unresponsive when used with Artemis/Aurora.

vjdato21 committed:
Updated maintainers

vjdato21 committed:
Fix keymapping issue

vjdato21 committed:
Enable `WAIT_FOR_USB`

vjdato21 committed:
Update readme.md

vjdato21 committed:
Update readme.md

vjdato21 committed:
Update readme.md
